src/App.tsx

```diff
@@ -1,7 +1,33 @@
 import React from 'react';
 
+class KeyListener extends React.Component {
+  state = {
+    pressedKey: '',
+  };
+
+  introText = 'The last pressed key is [';
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyPress);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyPress);
+  }
+
+  handleKeyPress = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: `${this.introText + event.key}]` });
```

Comment:
The pressedKey variable should contain only last pressed key, add this text when you display it
```suggestion
    this.setState({ pressedKey: event.key });
```